src/plugin/inputmethod.cpp

```diff
@@ -755,3 +755,19 @@ void InputMethod::onPluginPathsChanged(const QStringList& pluginPaths) {
 
     d->updateLanguagesPaths();
 }
+
+void InputMethod::showSystemSettings()
+{
+    auto previous = qgetenv("QT_WAYLAND_SHELL_INTEGRATION");
+    qunsetenv("QT_WAYLAND_SHELL_INTEGRATION");
```

Comment:
Why is this necessary? A comment to explain would be nice, as it's not intuitive as to why this would be needed.

Reply:
Addressed.